api/v1beta1/azurecluster_types.go

```diff
@@ -45,6 +45,11 @@ type AzureClusterSpec struct {
 	// +optional
 	BastionSpec BastionSpec `json:"bastionSpec,omitempty"`
 
+	// ControlPlaneEnabled enable control plane cluster components.
+	// +kubebuilder:default=true
+	// +optional
+	ControlPlaneEnabled bool `json:"controlPlaneEnabled,omitempty"`
```

Comment:
Unfortunately, there's no consistency across providers since this setting may vary.

e.g. on AWS:

```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
kind: AWSCluster
metadata:
  name: capi-quickstart
  namespace: default
spec:
  region: us-east-1
  sshKeyName: default
  controlPlaneLoadBalancer:
    loadBalancerType: disabled
```